### PR TITLE
Enabled Diag logging for DataCollector

### PR DESCRIPTION
**Pull request by harshjain2**
Enabled Diag logging for DataCollector.

Related to : https://github.com/microsoft/vstest/issues/309

### CODE REVIEW

File: src/Microsoft.TestPlatform.CoreUtilities/Helpers/CommandLineArgumentsHelper.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Helper class for processing argumets passed to a process.
+    /// </summary>
+    public class CommandLineArgumentsHelper
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CoreUtilities/Helpers/CommandLineArgumentsHelper.cs]: Does this need to be public? 
Nit: typo in docs

File: test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Helpers/CommandLineArgumentsHelperTests.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+{
+    using System.Collections.Generic;
+
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers;
+
+    [TestClass]
+    public class CommandLineArgumentsHelperTests
+    {
+        [TestMethod]
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Helpers/CommandLineArgumentsHelperTests.cs]: Add case for
I) null args
ii) --hello --world

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs [hunk 3]
```diff
@@ -216,7 +219,20 @@ private IList<string> GetCommandLineArguments(int portNumber)
             commandlineArguments.Add(PortOption);
             commandlineArguments.Add(portNumber.ToString());
 
+            if (!string.IsNullOrEmpty(EqtTrace.LogFile))
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: Tests

File: src/datacollector/Program.cs [hunk 4]
```diff
@@ -51,39 +57,35 @@ public static void Main(string[] args)
...
-        }
-
-        private static void Run()
-        {
-            var requestHandler = DataCollectionRequestHandler.Create(new SocketCommunicationManager(), new MessageSink());
 
             requestHandler.InitializeCommunication(port);
 
+            // Can only do this after InitializeCommunication because datacollector cannot "Send Log" unless communications are initialized
+            if (!string.IsNullOrEmpty(EqtTrace.LogFile))
```
Comment (singhsarab) [inline src/datacollector/Program.cs]: Check for acceptance test for diag.

**Review by singhsarab — APPROVED** [review]
Looks good with suggestions.

File: test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Helpers/CommandLineArgumentsHelperTests.cs (hunk 2 above)
Comment (harshjain2) [inline test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Helpers/CommandLineArgumentsHelperTests.cs]: done.

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs (hunk 3 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: done.

File: src/datacollector/Program.cs (hunk 4 above)
Comment (harshjain2) [inline src/datacollector/Program.cs]: implemented in acceptance tests for data collectors.

**Pull request merged**